client/pregame/CSavingScreen.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * CSavingScreen.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+
+#include "CSavingScreen.h"
+#include "SelectionTab.h"
+
+#include "../CPlayerInterface.h"
+#include "../gui/CGuiHandler.h"
+#include "../widgets/Buttons.h"
+#include "../widgets/TextControls.h"
+
+#include "../../CCallback.h"
+#include "../CGameInfo.h"
+#include "../../lib/CGeneralTextHandler.h"
+#include "../../lib/filesystem/Filesystem.h"
+#include "../../lib/CConfigHandler.h"
+
+CSavingScreen::CSavingScreen()
+	: CSelectionBase(CMenuScreen::saveGame)
+{
+	OBJ_CONSTRUCTION_CAPTURING_ALL;
+	bordered = false;
+	center(pos);
+	// MPTODO: I doubt saving screen need any of it
+	localSi = LOCPLINT->cb->getStartInfo();
+	localMi = new CMapInfo();
+	localMi->mapHeader = std::unique_ptr<CMapHeader>(new CMapHeader(*LOCPLINT->cb->getMapHeader()));
+
+	tabSel = new SelectionTab(screenType); //scenario selection tab
+	tabSel->recActions = DISPOSE; // MPTODO
+	tabSel->recActions = 255;
+	curTab = tabSel;
+
+	tabSel->onSelect = std::bind(&CSavingScreen::changeSelection, this, _1);
+	buttonStart = new CButton(Point(411, 535), "SCNRSAV.DEF", CGI->generaltexth->zelp[103], std::bind(&CSavingScreen::saveGame, this), SDLK_s);
+	buttonStart->assignedKeys.insert(SDLK_RETURN);
+}
+
+CSavingScreen::~CSavingScreen()
+{
+	vstd::clear_pointer(localSi);
```

Comment:
does CSavingScreen actually owns localSi?

Reply:
Just checked - yes it is. At least for now it's not unified.